biscuitVcfToBed.pl

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/perl
+
+use warnings;
+use strict;
+use IO::File;
+use List::MoreUtils qw(first_index);
+
+my %prev;
+
+my $mindepth=2;
+
+#handle gzipped or not vcfs
+my $file = $ARGV[0];
+if ($file =~ /.gz$/) {
+    open(INVCF, "gunzip -c $file |") || die "can’t open pipe to $file";
+} else {
+    open(INVCF, $file) || die "can’t open $file";
+}
+
+while(my $line = <INVCF>){
+    next if $line =~ /^#/;
+    chomp($line);
+    my @F = split("\t",$line);
+    next if $F[4] ne "."; #skip SNPs
+    next unless $F[7] =~ /CX=CG/; #we only care about CpGs
+
+    #get the CV and BT format field locations
+    my @fmt = split(":",$F[8]);
+    my $cv_ind = first_index { $_ eq 'CV' } @fmt;
+    my $bt_ind = first_index { $_ eq 'BT' } @fmt;
+
+    if ($bt_ind == -1 || $cv_ind == -1){
+        print STDERR "WARNING: expected CV and/or BT format fields not found:\n";
+        print STDERR $line . "\n";
+        next;
+    }
+    my @vals = split(":",$F[9]);
+
+    #cases
+    if(!(defined($prev{"chr"}))){
+        #prev is empty, this is a G (just print)
+        if($F[3] eq "G"){
+            print join("\t", ($F[0], $F[1]-1, $F[1], $vals[$cv_ind], $vals[$bt_ind])) . "\n"  if $vals[$cv_ind] >= $mindepth;
+        } elsif($F[3] eq "C"){
+            #prev is empty, this is a C (store)
+            $prev{"chr"} = $F[0];
+            $prev{"pos"} = $F[1];
+            $prev{"base"} = $F[3];
+            $prev{"CV"} = $vals[$cv_ind];
+            $prev{"BT"} = $vals[$bt_ind];
+        } else {
+            #not a C or G - this should never happen
+            die("ERROR: CpG context reported, but ref base isn't a C or G:\n$line\n");
+            next;
```

Comment:
Can take the `next` out after this die!  (I apparently don't have permissions to resolve the other conversations on this repo, but otherwise LGTM!)